src/components/FiltersModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import Modal from "react-modal";
+
+Modal.setAppElement("#root");
+
+type TFiltersModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  localFilters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  };
+  setLocalFilters: (filters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  }) => void;
```

Comment:
The argument here is of the same type as `localFilters`. So, it would make sense to define this as a standalone type and reuse it.

```suggestion
  localFilters: MyFiltersType;
  setLocalFilters: (filters: MyFiltersType) => void;
```